### PR TITLE
E2E MaxPool2d Bringup

### PR DESCRIPTION
**Pull request by LPanosTT**
Bringup maxpool2d e2e

### CODE REVIEW

File: forge/test/mlir/test_ops_tf.py [hunk 1]
```diff
@@ -97,3 +98,59 @@ def call(self, x):
...
+)
+def test_maxpool2d(
+    act_shape,
+):
+    tf.random.set_seed(0)
+
+    class MaxPool(tf.keras.Model):
+        def __init__(self):
+            super().__init__() 
+            self.pool = tf.keras.layers.MaxPool2D(pool_size=(2, 2), padding="valid", dtype=tf.bfloat16)
```
[the remark this answers is not in the document]
Comment (nvukobratTT) [reply forge/test/mlir/test_ops_tf.py:146]: Thanks for the info! Let's just add this node to the test file for now

[the remark this answers is not in the document]
Comment (LPanosTT) [reply forge/test/mlir/test_ops_tf.py:146]: I have :)

File: forge/csrc/forge_passes.cpp [hunk 2]
```diff
@@ -158,6 +158,16 @@ void run_optimization_graph_passes(graphlib::Graph *graph)
         passes::constant_folding(graph);
 
     passes::move_select_after_matmul_optional(graph);
+    
+    for (auto *node: graphlib::topological_sort(*graph)) {
```
[the remark this answers is not in the document]
Comment (nvukobratTT) [reply forge/csrc/forge_passes.cpp:162]: @LPanosTT regarding this fun, let's wrap it as a separate pass and check where is the best place to put it. 

E.g. for easier debugging, a good place for Nop removal can be before we emit the MLIR graph. To keep these Nops for debugging purposes. Also, if you decide to put it there, let's generate another reportify graph to have it explicit for a proper overview.

[the remark this answers is not in the document]
Comment (LPanosTT) [reply forge/csrc/forge_passes.cpp]: I've placed the pass in its own file, and called it in `pre_lowering_passes` at the very end, but right before the call to `recalculate_shapes`.

File: forge/csrc/passes/remove_nops.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+namespace tt::graphlib
+{
+class Graph;
+class OpNode;
+class Shape;
+}
+
+namespace tt::passes
+{
+// Returns true if any patterns were replaced with something commutable
```
Comment (nvukobratTT) [inline forge/csrc/passes/remove_nops.hpp:15]: Looks like invalid comment

**Review by nvukobratTT — APPROVED** [review]
LGTM! 🙌

**Comment by nvukobratTT** [comment]
Feel free to move it from Draft to Open PR @LPanosTT

**Pull request merged**